File: solidPrinciples/(L)Wrong.ts
class Bird {
  fly(): void {
    console.log("Flying...");
  }
}

class Penguin extends Bird {
  fly(): void {
    console.log("I can't fly!"); 
  }
}

function makeBirdFly(bird: Bird) {
  bird.fly();
}

const sparrow = new Bird();  
const penguin = new Penguin();

makeBirdFly(sparrow);  // Output: Flying...
makeBirdFly(penguin);  // Output: I can't fly! (unexpected behavior)

File: solidPrinciplesCpp/(L)Wrong.cpp
#include <iostream>

class Bird {
public:
    virtual void fly() {
        std::cout << "Flying...\n";
    }
};

class Penguin : public Bird {
public:
    void fly() override {
        std::cout << "I can't fly!\n";
    }
};

void makeBirdFly(Bird& bird) {
    bird.fly();
}

int main() {
    Bird sparrow;
    Penguin penguin;

    makeBirdFly(sparrow);  // Expected: "Flying..."
    makeBirdFly(penguin);  // Problematic: Violates LSP because Penguins can't fly
}
